br/pkg/membuf/buffer.go

```diff
@@ -172,17 +175,51 @@ func (p *Pool) NewBuffer(opts ...BufferOption) *Buffer {
 	return b
 }
 
-// Reset resets the buffer, the memory is still retained in this buffer.
+// smallObjOverheadBatch is the batch size to acquire memory from limiter. 256KB
+// can store 256KB/24B = 10922 []byte objects, or 256KB/12B = 21845 SliceLocation
+// objects.
+const smallObjOverheadBatch = 256 * 1024
+
+// recordSmallObjOverhead records the memory cost of []byte or SliceLocation into
+// pool's limiter. The caller will ensure the pool's limiter is not nil.
+func (b *Buffer) recordSmallObjOverhead(n int) {
+	if n > b.smallObjOverheadCache {
```

Comment:
seems `n` never bigger than it, and b.smallObjOverheadCache always <= 0?

Reply:
the initial value of `smallObjOverheadCache` is `0`. For the first invocation `n` is bigger than it and will go through line 187~189. Then the next few invocations `n` is less than `smallObjOverheadCache` so directly goes to line 191.

Reply:
my fault, i read it as `if n > smallObjOverheadBatch {` 😓